### DISCUSSION TITLE
is there any Java API without TSDataType to insert records into an existed timeseries?

### DISCUSSION THREAD

**Discussion by holtenko** [apache/iotdb#5352] [Q&A] [answered]
I have created some timeseries, and the TSDataType was defined during timeseries creating. I think it's not necessary to input TSDataType when I call API to insert records.
So, is there any Java API without TSDataType to insert records into an existed timeseries? Thank you.

**Comment by holtenko** [comment]
Just like: `insertRecord(String deviceId, long time, List<String> measurements, List<Object> values)`

**Comment by HTHou** [comment]
https://github.com/apache/iotdb/blob/b1033855571a534bb30d6375150fdf395ca3dbc6/example/session/src/main/java/org/apache/iotdb/SessionExample.java#L292-L307

**Reply by holtenko** [reply]
Hi @HTHou , thanks for your answer.
I checked the doc(https://iotdb.apache.org/UserGuide/Master/API/Programming-Java-Native-API.html#insert-with-type-inference) of this method, it seems will cost some time to do type inference.
For the existed timeseries, does it still have the time cost for type inference?

**Reply by HTHou** [reply]
Yes, it will take some time to do type inference. 

Actually, I think it is possible to add a Syntactic sugar about what you want. Just using the actual Object type of value to infer the TsDataType in the Session, and then send the values and TsDataTypes to server.

**Comment by HTHou** [comment, marked as answer]
https://issues.apache.org/jira/browse/IOTDB-2815 
An issue has been created, welcome to be a contributor of Apache IoTDB!

**Reply by holtenko** [reply]
Thank you @HTHou 
I will try to contribute for this feature.
